frodo.js

```diff
@@ -1,75 +1,68 @@
-/**
```

Comment:
SCRIPT-Well done with the efficient error handling, it covers all the possible scenarios.